src/projectscene/internal/projectsceneconfiguration.cpp

```diff
@@ -156,3 +170,87 @@ muse::async::Channel<ClipStyles::Style> ProjectSceneConfiguration::clipStyleChan
 {
     return m_clipStyleChanged;
 }
+
+StereoHeightsPref::AsymmetricStereoHeights
+ProjectSceneConfiguration::stereoHeightsPref() const
+{
+    return muse::settings()->value(STEREO_HEIGHTS_PREF).toEnum<StereoHeightsPref::AsymmetricStereoHeights>();
+}
+
+void ProjectSceneConfiguration::setStereoHeightsPref(
+    StereoHeightsPref::AsymmetricStereoHeights pref)
+{
+    if (stereoHeightsPref() == pref) {
+        return;
+    }
+    muse::settings()->setSharedValue(STEREO_HEIGHTS_PREF, muse::Val(pref));
+}
+
+muse::async::Notification ProjectSceneConfiguration::stereoHeightsPrefChanged() const
+{
+    return m_asymmetricStereoHeightsChanged;
+}
+
+std::vector<std::string> ProjectSceneConfiguration::asymmetricStereoHeightsWorkspaces() const
+{
+    // workspaces that have asymmetricStereoHeights enabled are stored as a string in muse::settings
+    // "Classic|Music|Advanced audio editing"
+    std::vector<std::string> result;
+    std::string combinedString = muse::settings()->value(ASYMMETRIC_STEREO_HEIGHTS_WORKSPACES).toString();
+    std::stringstream ss(combinedString);
+    std::string item;
+
+    while (std::getline(ss, item, '|')) {
+        result.push_back(item);
+    }
+
+    return result;
+}
+
+void ProjectSceneConfiguration::setAsymmetricStereoHeightsWorkspaces(std::vector<std::string>& asymmetricWorkspaces)
+{
+    if (asymmetricStereoHeightsWorkspaces() == asymmetricWorkspaces) {
+        return;
+    }
+
+    // convert vector to single string saveable in muse::settings
+    std::ostringstream oss;
+    for (size_t i = 0; i < asymmetricWorkspaces.size(); ++i) {
+        if (i > 0) {
+            oss << '|';
+        }
+        oss << asymmetricWorkspaces.at(i);
+    }
+
+    std::string result = oss.str();
+
+    muse::settings()->setSharedValue(ASYMMETRIC_STEREO_HEIGHTS_WORKSPACES, muse::Val(result));
+}
+
+muse::async::Notification ProjectSceneConfiguration::asymmetricStereoHeightsWorkspacesChanged() const
+{
+    return m_asymmetricStereoHeightsWorkspacesChanged;
+}
+
+void ProjectSceneConfiguration::asymmetricStereoHeightWorkspacesCleanUp()
+{
+    // cleanup after user removes workspace
+    std::vector<std::string> asymmetricWorkspaces = asymmetricStereoHeightsWorkspaces();
+    auto workspaces = workspaceManager()->workspaces();
```

Comment:
Please don't use managers/services/etc in configurations
I think we can fill the list of workspaces directly in the model, yes?